docs/architecture/ADR-006-oracle.md

```diff
@@ -0,0 +1,59 @@
+# ADR 006: Oracle
+
+## Changelog
+
+- November 29th, 2021: Initial Draft (@adamewozniak)
+
+## Status
+
+Proposed
+
+## Context
+
+Umee needs an oracle to determine the price of assets.
+
+From section 5.1 of the [Umee Whitepaper](https://umee.cc/umee-whitepaper/):
+
+> Oracle reads asset price and updates the state to the Asset Facility Coordination
+Protocol
+
+We've already decided to import a large chunk of this from [Terra's oracle](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-oracle.html), although a few parts of this are specific to Terra's protocol and do not need to be implemented.
+
+## Alternative Approaches
+
+- Cloning the x/oracle module completely. This would leave our code dirty, and we'd later have issues interfacing with Terra's [Cross Exchange Rate](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-oracle.html#compute-cross-exchange-rate-using-reference-terra), since it's designed for getting the exchange rate of only Terra.
+- Using something like Band or Chainlink. This would be additional overhead, and we'd [have less control](https://github.com/umee-network/umee/issues/97#issuecomment-923914840) over how our oracle works.
+
+## Decision
+
+We'd like to use the concepts introduced in [Terra's Oracle](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-oracle.html#concepts), but with a few modifications :
+
+- Combine the `price-feeder` and `feeder` into a single binary [Ref](https://github.com/umee-network/umee/issues/97#issuecomment-939610302)
+- Only support MsgAggregateExchangeRate(Pre)Vote, i.e. not allow individual price updates [Ref](https://github.com/umee-network/umee/issues/97#issuecomment-939610302)
+- Skip the logic for the [Cross Exchange Rate](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-oracle.html#compute-cross-exchange-rate-using-reference-terra), and record simplified exchange rates like `ATOMUSDT`
+- Remove support for the [Tobin Tax](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-market.html#tobintax), a Terra-specific fee for when users spot trade.
+
+## Detailed Design
+
+- Terra's design for the voting procedure as documented [here](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-oracle.html#voting-procedure)
+
```

Comment:
```suggestion

## API

The `x/oracle` module will provide the following methods on its keeper, to be used by `x/leverage`:

```go
    // example - we should discuss if this is the best way to do things
    GetValue(asset sdk.Coin) (sdk.Dec, error) // get the USD value of an input sdk.Coin
    GetTotalValue(asset sdk.Coin) (sdk.Dec, error) // return the total USD value of all coins in an sdk.Coins
```
```